### DISCUSSION TITLE
Search multiple columns for plain text

### DISCUSSION THREAD

**Discussion by saltcod** [supabase/supabase#2031] [Questions] [answered]
Hey all!  LOVING Supabase so far! Heaven! 

I've got a table with 2 columns — name and company— I want to search on and I'm not sure the right way to do that. 

The closest I've been able to get is:

`const { data, error, range, count } = await supabase.from('earners')
			.select('*', { count: 'exact' })
			.textSearch('name', 'gates', {type: 'phrase'})`

This search successfully brings up "Bill Gates" in the name column.  But this won't:

`const { data, error, range, count } = await supabase.from('earners')
			.select('*', { count: 'exact' })
			.textSearch('name', 'gate', {type: 'phrase'})`

So I'm looking for a kind of "fuzzy" search across 2 columns (name, company).  I'd like for any of these terms to give me a result for Bill Gates —  "bill", "gates", "gate", "Gate",  "microsoft", "Microsoft", ...etc...

Basically, I don't know much about search yet =) 

Any help much appreciated!

**Comment by soedirgo** [comment, marked as answer]
Hey @saltcod, I'm guessing the values in `name` and `company` are short in which case [`ilike`](https://supabase.io/docs/reference/javascript/ilike) might be more appropriate. And to match the phrase on either columns you'll need [`or`](https://supabase.io/docs/reference/javascript/or):
```js
const phrase = 'gate'
const { data, error } = await supabase
    .from('earners')
    .select()
    .or(`name.ilike.*${phrase}*,company.ilike.*${phrase}*`)
```

**Reply by saltcod** [reply]
Sweeeeeeeeeeeeeet!   Thanks @soedirgo !! Brilliant.  V. much appreciated!

**Comment by saendu** [comment]
@saltcod this should work now with type websearch
```
.textSearch('name', 'gate', {
          type: 'websearch',
          config: 'english'
        })
```
